core/services/keystore/csa.go

```diff
@@ -133,6 +134,26 @@ func (ks *csa) Export(id string, password string) ([]byte, error) {
 	return key.ToEncryptedJSON(password, ks.scryptParams)
 }
 
+// EnsureKey verifies whether the CSA key has been created, if not, it creates it.
+func (ks *csa) EnsureKey() (csakey.KeyV2, bool, error) {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	if ks.isLocked() {
+		return csakey.KeyV2{}, false, ErrLocked
+	}
+
+	if len(ks.keyRing.CSA) > 0 {
+		return csakey.KeyV2{}, true, nil
```

Comment:
we should return the first key here, rather than a generic empty key

Reply:
Actually, non of the `ensure` methods we have on the keystores are retuning anything on this specific check.

Reply:
why wouldn't this check also throw `ErrCSAKeyExists`?

Reply:
in this specific case, the `Ensure` signature I've seen in all the keystores is returning always 3 pieces of information, in which the boolean value is the one that states whether a key is already stored. So doesn't really feel necessary to return an error when the consumer is expecting a boolean that tells whether the key exists or not...

could be wrong.

Reply:
hmm yea was just thinking it solves Ryan's concern with returning the first key (instead of returning the nil/empty struct like on other error scenarios).  Also everywhere else with a check on the number of keys returns the `ErrCSAKeyExists` for multiple keys.

But yea I really don't have the answer.  I do think that the change reads more consistently if this check would return an `ErrCSAKeyExists`

Reply:
I would too prefer to return an existing key. On the receiving side, you are logging the ID of this key.

Reply:
To follow up with your thoughts, then I think we should update all the keystores existent `Ensure` signatures to reflect this approach, if everybody is fine with it. Cause right now, returning `true` and `nil` in this check for `Ensure` methods, is the current "standard", as far as I've seen.

cc: @RyanRHall @jmank88

Reply:
If we moved the `Created X key with ID Y` logging inside of these methods, then we could simplify the return to only an `error` and eliminate this complexity entirely.

Reply:
I agree with @jmank88

Reply:
Might want to take care not to log while holding the lock, but otherwise should simplify a lot overall.

Reply:
Oh just realized, you are logging the key  ID only if it was newly created.
But generally, if we have existing key, we should return it here, instead returning an empty key.

Reply:
I will also change the signatures to not return keys, since the keys are only used for logging.

Reply:
yikes, looks like the only keystore that handles the `Ensure` function correctly is the eth key store. The ensure method should return the key if one already exists.